### PR TITLE
vendor containers/image

### PR DESCRIPTION
**Pull request by runcom**
specifically https://github.com/containers/image/pull/115 and also add an integration test

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
@mtrmac PTAL

**Comment by runcom** [comment]
I think it's time to setup the schema 1 registry in tests :) I'll do it

**Comment by mtrmac** [comment]
If it’s all the same to you, I’d prefer running the tests against a v2 registry with s2; s1 images’ signatures eventually _should_ break with fixed s1 upload (#88), so converting most of the tests from Docker Hub estesp (s1+s2+list) against OpenShift to s2 against v2 registry would be nicer.

**Comment by runcom** [comment]
> If it’s all the same to you, I’d prefer running the tests against a v2 registry with s2; s1 images’ signatures eventually should break with fixed s1 upload (#88), so converting most of the tests from Docker Hub estesp (s1+s2+list) against OpenShift to s2 against v2 registry would be nicer.

wait, v2s2 broke something in openshift here, I'll revert some changes and you can see

**Comment by runcom** [comment]
@mtrmac do you know why the current test with `docker://busybox:latest` which is v2s2 will make atomic registry fails with 500?

**Comment by mtrmac** [comment]
Atomic Registry by default refuses to accept s2.  https://github.com/mtrmac/skopeo/tree/docker-push-to-tag enables that, but then we are running tests in a non-default configuration, not a great idea.

That’s why I was suggesting a docker/distribution registry + s2, instead of Atomic. (And why this is so much work.)

**Comment by runcom** [comment]
To summarize, you want that test to push the image to an in-test docker/distribution V2 registry, instead of the atomic registry, right?

**Comment by mtrmac** [comment]
At least `TestCopySignatures`, yes; perhaps others may be needed. There should still be _something_ testing against the Atomic Registry.

**Comment by runcom** [comment]
Right, we can find an s1 image to still test _something_ for the atomic registry, fine. I'll finish this tomorrow morning. The v2s2 registry is already setup in the test infra so it shouldn't be _that_ hard I hope.

**Comment by mtrmac** [comment]
Yes, at least `TestCopyDockerSigstore` is already using it.

**Comment by runcom** [comment]
@mtrmac tests fixed, PTAL

File: integration/copy_test.go [hunk 1]
```diff
@@ -15,17 +15,24 @@ import (
...
 
 type CopySuite struct {
+	ss       *SkopeoSuite
 	cluster  *openshiftCluster
 	registry *testRegistryV2
 	gpgHome  string
 }
 
+func (s *CopySuite) SetUpTest(c *check.C) {
+	s.ss.SetUpTest(c)
```
Comment (mtrmac) [inline integration/copy_test.go]: `SkopeoSuite` is setting up a fresh set of _five_ registries for _every single test_.  That seems a fair bit of unnecessary overhead, especially when `CopySuite.SetUpSuite` is already setting up a v2 registry.

File: integration/copy_test.go [hunk 2]
```diff
@@ -107,13 +118,33 @@ func (s *CopySuite) TestCopySimple(c *check.C) {
 
 	// FIXME: It would be nice to use one of the local Docker registries instead of neeeding an Internet connection.
 	// "pull": docker: → dir:
-	assertSkopeoSucceeds(c, "", "copy", "docker://estesp/busybox:latest", "dir:"+dir1)
+	assertSkopeoSucceeds(c, "", "copy", "docker://estesp/busybox:amd64", "dir:"+dir1)
```
Comment (mtrmac) [inline integration/copy_test.go]: Heh :) Nice and small.

File: integration/copy_test.go (hunk 1 above)
Comment (runcom) [inline integration/copy_test.go]: fair.

**Comment by runcom** [comment]
Should be good to go if tests agree @mtrmac

**Pull request merged**